src/Controller/MovieController.php

```diff
@@ -21,19 +21,43 @@ public function __construct(
     }
 
     #[Route('/movies', methods: ['GET'], name: 'homepage')]
-    public function homePage(#[MapQueryParameter] string $page = '1')
-    {
-        if (!ctype_digit($page)) {
-            throw new BadRequestHttpException("Query parameter 'page' is missing or invalid");
+    public function homePage(
+        #[MapQueryParameter] string $page = '1',
+        #[MapQueryParameter] string|null $title = null,
+    ) {
+        if (!ctype_digit($page) || $page < 1) {
+            throw new BadRequestHttpException("Query parameter 'page' is invalid");
+        }
+
+        $paginatedResult = $this->movieRepository->findPaginated($page, self::MOVIES_PER_PAGE, $title);
+
+        // If there are no movies in paginated result
+        if (count($paginatedResult) === 0 ) {
```

Comment:
I am not sure if you need a whole new page for "no results".
You can check in twig if the total is 0 or not and based on that display the grid and pagination.